lib/ddq-backend-mock.js

```diff
@@ -0,0 +1,304 @@
+         * @return {(Ddq~record|null)} not finding a record returns nothing
+         */
+        getRecord(recordId) {
+            if (this.storedData[recordId]) {
+                return this.storedData[recordId];
+            }
+
+            return null;
+        }
+
+
+        /**
+         * Looks to see if there are any messages in the storeData and creates
+         * the wrapped message.
+         *
+         * @return {(Ddq~wrappedMessage|null)}
+         */
+        getWrappedMessage() {
+            var file, files, keys;
+
+            files = this.readData();
+            keys = Object.keys(files);
+
+            if (keys.length) {
+                // eslint-disable-next-line no-bitwise
+                file = files[keys[keys.length * Math.random() << 0]];
+
+                if (file && !file.isProcessing) {
+                    files[file.id].isProcessing = true;
+                    this.writeRecord(file);
+
+                    return {
+                        heartbeat: heartbeat.bind(null, this, file.id),
+                        heartbeatKill: heartbeatKill.bind(null, this.file.id),
+                        message: file.message,
+                        remove: remove.bind(null, this, file.id),
+                        requeue: requeue.bind(null, this, file.id)
+                    };
+                }
+            }
+
+            return null;
+        }
+
+
+        /**
+         * Hey Listen!
+         * Initiates the polling.
+         */
+        listen() {
+            if (!this.config.noPolling) {
+                this.startPolling();
+            }
+        }
+
+
+        /**
+         * For times when polling needs to pause so no more messages are sent.
+         */
+        pausePolling() {
+            timers.clearInterval(this.poller);
+            this.poller = null;
+        }
+
+
+        /**
+         * Gets the stored data or a particular one beind searched for.
+         *
+         * @return {Object} all records
+         */
+        readData() {
+            return this.storedData;
+        }
+
+
+        /**
+         * Resumes, starts polling.
+         */
+        resumePolling() {
+            this.startPolling();
+        }
+
+
+        /**
+         * Sends a record off to storage.
+         *
+         * @param {*} message
+         * @param {Function} callback
+         */
+        sendMessage(message, callback) {
+            var finalMessage;
+
+            if (this.config.sendFail) {
+                callback(new Error("Could not send"));
+            } else {
+                finalMessage = {};
+                finalMessage.heartbeatTime = null;
+                finalMessage.id = crypto.createHash("sha256").update(message).digest("hex");
+                finalMessage.isProcessing = false;
+                finalMessage.message = message;
+                finalMessage.owner = null;
+                finalMessage.requeued = false;
+                this.writeRecord(finalMessage);
```

Comment:
Note, this STAYS but you could just replace this line with `files[finalMessage.id] = finalMessage`;